cmd/juju/controller/unregister.go

```diff
@@ -102,10 +101,20 @@ func (c *unregisterCommand) Run(ctx *cmd.Context) error {
 		return errors.Trace(err)
 	}
 
-	if !c.assumeYes {
+	if !c.assumeNoPrompt {
+		assumeNoPrompt, skipErr := jujucmd.CheckSkipConfirmEnvVar()
+		if skipErr != nil && !errors.Is(skipErr, errors.NotFound) {
+			return errors.Trace(skipErr)
+		} else {
+			c.assumeNoPrompt = assumeNoPrompt
+		}
+	}
```

Comment:
Unfortunately unregister doesn't inherit from destroyCommandBase so this will need to be duplicated in this command's init

But perhaps it should? worth a look. There's a fair amount in common between all 3 commands, perhaps destroyCommandBase and unregisterCommand should inherit from some new base, with the flags, maybe the deprecation warning, etc.

Probably more work than it's worth